### PR TITLE
Use setState() instead of mutating state directly.

### PR DESCRIPTION
**Pull request by peacechen**
https://facebook.github.io/react/docs/react-component.html#instance-properties
Also fixes corner case of modal opening immediately after closing.
See #174

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -321,8 +339,9 @@ var ModalBox = createReactClass({
       inSwipeArea = false;
       if (this.props.entry === 'top' ? -state.dy > this.props.swipeThreshold : state.dy > this.props.swipeThreshold)
         this.animateClose();
-      else
+      else if (!this.state.isOpen) {
```
Comment (peacechen) [inline index.js]: This check needs to happen to prevent the modal from opening immediately after closing in specific situations.

**Comment by peacechen** [comment]
@maxs15  Would you please review this for merging?

**Comment by superandrew213** [comment]
@maxs15 can you merge this?

**Comment by gvlekke** [comment]
merged in https://github.com/gvlekke/react-native-modalbox/, im also going to merge some other pull requests.

**Pull request merged**